test/e2e/fetch_target_logs.go

```diff
@@ -0,0 +1,127 @@
+package e2e
+
+import (
+	"context"
+	"flag"
+	"path/filepath"
+	"os"
+	"io/ioutil"
+	"io"
+	"bytes"
+	"log"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/kubectl/pkg/describe"
+)
+
+func FetchTargetLogs() {
+	dirName := "/tmp/target_cluster_logs2"
+
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	//Get all namespaces
+	namespaces, err := clientset.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		log.Fatal(err)
+	}
+	
```

Comment:
Would you add here also equivalent of `kubectl --kubeconfig .... get pods -A` before namespace iteration, we've been missing that.

Follow-up PR is also OK, but since you want to get this merged before vacations, adding it here makes sense.